extended/src/main/java/apoc/util/ExtendedUtil.java

```diff
@@ -187,10 +191,10 @@ private static Object convertValue(String value, String typeName) {
                 return null;
             default:
                 // If ends with "Array", for example StringArray
-                if (typeName.endsWith("Array")) {
+                if (typeName.endsWith("Array") || typeName.startsWith("List")) {
                     value = StringUtils.removeStart(value, "[");
                     value = StringUtils.removeEnd(value, "]");
-                    String array = typeName.replace("Array", "");
+                    String array = WordUtils.capitalize(typeName.replace("Array", "").replace("List", ""));
```

Comment:
Facendo la modifica di cui sopra, dovresti cambiare anche "Array" e "List" in "array" e "list", se non sbaglio